Lesson2/Holodenko_Artem/task_2/main.cpp

```diff
@@ -0,0 +1,175 @@
+#include <iostream>
+#include <queue>
+#include <vector>
+#include <math.h>
+
+struct node
+{
+    int n = 0;
+    struct node *left = nullptr;
+    struct node *right = nullptr;
+    struct node *level = nullptr;
+};
+
+
+std::vector<node> get_data()
+{
+    int number_levels, number_nodes=0;
+    std::cout << "Enter number of levels: ";
+    std::cin >> number_levels;
+
+    //кількість вузлів
+    for (int i=0; i<number_levels; i++) number_nodes+=pow(2, i);
+
+    //std::cout << number_nodes << std::endl;
+    std::vector<node> data_vector (number_nodes);
+
+    //заповнення дерева, 0 пустий вузол
+    for (int i=0; i<number_nodes; i++)
+    {
+        std::cout << "Enter node " << i+1 << ": ";
+        std::cin >> data_vector[i].n;
+    }
+
+    //створення зв'язків
+    {
+        int m, n;
+        int level;
+        for (int i=0; i<number_nodes; i++)
+        {
+            level = 0;
+            m=i+1;
+            while (m>pow(2, level))
+            {
+                m-=pow(2, level);
+                level++;
+            }
+            level++;
+
+            //std::cout << "level " << level << std::endl;
+            //std::cout << "m " << m << std::endl;
+
+            n = 0;
+            for (int j=0; j<level; j++) n+=pow(2, j);
+            n+=(m-1)*2;
+
+            //std::cout << "n1 " << n << std::endl;
+
+            if (n<number_nodes and data_vector[n].n!=0)
+            {
+                if (data_vector[i].n==0)
+                {
+                    data_vector[n].n=0;
+                }
+                else
+                {
+                    data_vector[i].left=&data_vector[n];
+                }
+            }
+
+            n = 0;
+            for (int j=0; j<level; j++) n+=pow(2, j);
+            n+=(m-1)*2+1;
+
+            //std::cout << "n2 " << n << std::endl;
+
+            if (n<number_nodes and data_vector[n].n!=0)
+            {
+                if (data_vector[i].n==0)
+                {
+                    data_vector[n].n=0;
+                }
+                else
+                {
+                    data_vector[i].right=&data_vector[n];
+                }
+            }
+        }
+    }
+
+    return  data_vector;
+}
+
+
+
+
+
+void print_node(std::vector<node> &data_vector)
+{
+    for (int i=0; i<data_vector.size(); i++)
+    {
+        if (data_vector[i].n == 0) continue;
+        std::cout << "node " << data_vector[i].n << ":" <<std::endl;
+        if (data_vector[i].left!=nullptr)
+        {
+            std::cout << "- " << data_vector[i].left->n << std::endl;
+        }
+        if (data_vector[i].right!=nullptr)
+        {
+            std::cout << "- " << data_vector[i].right->n << std::endl;
+        }
+        if (data_vector[i].level!=nullptr)
+        {
+            std::cout << "- " << data_vector[i].level->n << std::endl;
+        }
+    }
+}
+
+
+void func(std::queue<node*> data_queue)
```

Comment:
**also, you didn't got the task again**
Queue should be used not as input data, it should help you to solve the problem.
Please rework two tasks.